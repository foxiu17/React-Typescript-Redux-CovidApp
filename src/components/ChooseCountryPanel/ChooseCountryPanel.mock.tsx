import { ICountries } from "../../Interfaces/FetchedDataInterfaces";

export const MockCountries: Array<ICountries> = [
  {
    Country: "ALA Aland Islands",
    Slug: "ala-aland-islands",
    ISO2: "AX"
  },
  {
    Country: "Afghanistan",
    Slug: "afghanistan",
    ISO2: "AF"
  }
];