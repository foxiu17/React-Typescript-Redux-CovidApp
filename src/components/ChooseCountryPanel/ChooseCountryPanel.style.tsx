import styled from 'styled-components';

import UI_TEXTFIELD from '@material-ui/core/TextField';

import { Grid as commonGrid } from '../../Assets/Common/Containers.style';
import { Button as commonButton } from '../../Assets/Common/Buttons.style';

export const Grid = styled(commonGrid)`
  width: 100%;
  height: 100%;
`;

export const TextField = styled(UI_TEXTFIELD)``;

export const Button = styled(commonButton)``;