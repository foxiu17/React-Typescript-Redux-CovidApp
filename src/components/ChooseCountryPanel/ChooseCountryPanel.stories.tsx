import React from "react";

import ChooseCountryPanel from "./ChooseCountryPanel";

import { MockCountries } from "./ChooseCountryPanel.mock";

export default {
  component: ChooseCountryPanel,
  title: "ChooseCountryPanel"
};

// @ts-ignore
export const initial = () => <ChooseCountryPanel countries={MockCountries}  currentValue="" />;