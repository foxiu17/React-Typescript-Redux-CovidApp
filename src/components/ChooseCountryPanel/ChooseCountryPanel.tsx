import React, {useEffect} from "react";

import { ICountryPanelProps } from "../../Interfaces/Interfaces";
import { ICountries } from "../../Interfaces/FetchedDataInterfaces";

import { Grid, TextField, Button } from "./ChooseCountryPanel.style";

const ChooseCountryPanel = ({ countries, handleChange, currentValue, submitFunc }: ICountryPanelProps) => {
  useEffect(() => {
    countries.sort(function(a: ICountries, b: ICountries) {
      if (a.Country < b.Country) return -1;
      if (a.Country > b.Country) return 1;
      return 0;
    });
  });
  return (
    <Grid container>
      <Grid item xs={12}>
        <TextField
          select
          label="Native select"
          value={currentValue}
          onChange={handleChange}
          SelectProps={{
            native: true,
          }}
          helperText="Please select your currency"
          variant="outlined"
        >
          {countries.map((option: ICountries) => (
            <option key={option.ISO2} value={option.Slug}>
              {option.Country}
            </option>
          ))}
        </TextField>
      </Grid>
      <Grid item xs={12}>
        <Button type="button" onClick={submitFunc}>Submit</Button>
      </Grid>
    </Grid>
  )
};

export default ChooseCountryPanel;