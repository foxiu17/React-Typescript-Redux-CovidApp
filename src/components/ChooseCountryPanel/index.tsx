import ChooseCountryPanel from "./ChooseCountryPanel";

export default ChooseCountryPanel;