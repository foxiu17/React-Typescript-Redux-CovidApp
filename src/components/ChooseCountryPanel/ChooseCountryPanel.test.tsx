import React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from "enzyme";

import ChooseCountryPanel from "./ChooseCountryPanel";


describe('Test ChooseCountryPanel component', () => {
  describe('Render tests', () => {
    test('Render component without crashing', () => {
      const submitFunc= jest.fn();
      const handleChange = jest.fn();
      const component: ShallowWrapper = shallow(<ChooseCountryPanel countries={[]} handleChange={handleChange} submitFunc={submitFunc} currentValue='' />);

      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    })
  })
})