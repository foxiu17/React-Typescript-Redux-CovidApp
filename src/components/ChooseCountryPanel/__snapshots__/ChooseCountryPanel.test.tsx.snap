// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ChooseCountryPanel component Render tests Render component without crashing 1`] = `ShallowWrapper {}`;
